src/Build/BackEnd/BuildManager/BuildManager.cs

```diff
@@ -2184,20 +2184,13 @@ private void OnThreadException(Exception e)
                             continue;
                         }
 
-                        submission.CompleteLogging(false);
-
                         // Attach the exception to this submission if it does not already have an exception associated with it
-                        if (submission.BuildResult?.Exception == null)
+                        if (!submission.IsCompleted && submission.BuildResult != null && submission.BuildResult.Exception == null)
```

Comment:
Doesn't `submission.BuildResult?.Exception == null` mean the same thing as `submission.BuildResult != null && submission.BuildResult.Exception == null`? [SharpLab](https://sharplab.io/#v2:C4LglgNgPgAgTARgLACgYGYAE9MGFMDeqmJ2WMALJgLIAUAlIcaSwMoCuARgLZgDOfMAHsAdpj5deA4WIC8mEQFMA7pg49+g0QwDczFiX0GwAM0y0AhBI3TRAOgCSfXEO4AHCIuCKAJpgBk/uKSmjJ2AELskD4ASooSEMCYFvIi7BAQAUHWUloiEVEQsfHpwHYAogAeAMaKbsAymLKp6RD0RiwA9J2YpuZWIbb5Ti7unt5+gcE2eQXRcQnAAPwVNXUNok0tGe0oBqREe/uYAL5GZygXGNhwaoN5qIcs15xCQpkjrh5evnpHJNdIvMSolMECigtSn8ruRbuDiotHkZrlVavVGqj1jJoaggA==) shows an IL difference but it appears to just not be loading the `BuildResult` field twice in the `?.` version.

Reply:
If submission.BuildResult is null, submission.BuildResult?.Exception is null; `?` is like `<thing before> == null ? null : <thing before>.<thing after>`.

(I ran [this](https://sharplab.io/#v2:C4LgTgrgdgPgAgJgIwFgBQcDMACR2DC2A3utmbggsaebXEgAzYAO2AvNlBADbcDcNWmXoBOABTMA/ADoAcgHswAWwCG3AJYAvAKZiAlOw5deegWloBfdFbRA).)

Reply:
With value type fields, one can write something like `submission.BuildResult?.SomeFlag == false` and this is equivalent to `submission.BuildResult != null && submission.BuildResult.SomeFlag == false`. But it doesn't work with reference type fields as the type of `submission.BuildResult?.Exception` is just `Exception` and if it evaluates to `null`, you can't tell which of the two is `null`.

Reply:
You're right; it's the null-to-bool comparison that saves the former case which doesn't apply here.